client.go

```diff
@@ -299,9 +340,23 @@ func (r *marathonClient) apiCall(method, path string, body, result interface{})
 
 		// step: check for a successfull response
 		if response.StatusCode >= 200 && response.StatusCode <= 299 {
+
 			if result != nil {
-				if err := json.Unmarshal(respBody, result); err != nil {
-					return fmt.Errorf("failed to unmarshal response from Marathon: %s", err)
+				// If we have a deployment ID header and no response body, give them that
+				deploymentID := response.Header.Get(deploymentHeader)
+				if len(respBody) == 0 && deploymentID != "" {
+					d := DeploymentID{
+						DeploymentID: deploymentID,
+					}
+					switch result.(type) {
+					case *DeploymentID:
+						*result.(*DeploymentID) = d
```

Comment:
We could postpone the definition of `d` until we're clear that `result` is of the right type, or even just assign the `DeploymentID` struct inline here.